Serviceaccounts: Add serviceaccount docs (#41712)
grafana/grafana: docs/sources/http_api/serviceaccount.md

+++
title = "Service account HTTP API"
description = "Grafana Service account HTTP API"

> keywords = ["grafana", "http", "documentation", "api", "serviceaccount"]

Reviewer: ```suggestion
keywords = ["grafana", "http", "documentation", "api", "service account"]
```

Reviewer: Actually, I am not sure about change suggestion, given that the API path is `serviceaccount` and the documentation is `service account`. The former is harder to read.

Author: its a good point. I am not sure here if we want to separate the service account. Sorry to always point to google cloud, but here they separate service and account(s) in their documentation.

Maybe we should too?
https://console.cloud.google.com/iam-admin/serviceaccounts

Author: they also note plural and not singluar

Author: these keywords btw, can they be separate with a space?

Author: thx for the input

aliases = ["/docs/grafana/latest/http_api/serviceaccount/"]
+++

# Service account API

This API allows you to interact programmatically with the [Service accounts]({{< relref "../manage-users/serviceaccount/_index.md" >}}).

> > If you are running Grafana Enterprise and have [Fine-grained access control]({{< relref "../enterprise/access-control/_index.md" >}}) enabled, for some endpoints you would need to have relevant permissions.
> > Refer to specific resources to understand what permissions are required

Reviewer: ```suggestion
** > Note:** If you are using Grafana Enterprise and have [Fine-grained access control]({{< relref "../enterprise/access-control/_index.md" >}}) enabled, for some endpoints you need to have relevant permissions. Refer to specific resources to understand what permissions are required.
```

## Delete a service account

`DELETE /api/serviceaccounts/:serviceaccountId`

#### Required permissions

> See note in the [introduction]({{< ref "#user-api" >}}) for an explanation.

Reviewer: ```suggestion
For details, see the [introduction]({{< ref "#user-api" >}}).
```

Reviewer: @achatterjee-grafana, do all references need to use `relref`? @GrafanaWriter and @chri2547, this question is something that we ought to include in our Markdown conventions at https://github.com/grafana/grafana/blob/main/contribute/style-guides/documentation-markdown-guide.md.

Reviewer 2: For internal content within the same repo, yes.

| Action                 | Scope              |
| ---------------------- | ------------------ |
| serviceaccounts:delete | serviceaccounts:\* |

Deletes the given service account if it exists.

> **Example Request**:

Reviewer: ```suggestion
**Example request**:
```

```http
DELETE /api/serviceaccounts/1 HTTP/1.1
Accept: application/json
Content-Type: application/json
Authorization: Bearer eyJrIjoiT0tTcG1pUlY2RnVKZTFVaDFsNFZXdE9ZWmNrMkZYbk
```

> **Example Response**:

Reviewer: ```suggestion
**Example response**:
```
